CoreAlgorithms/src/au/gov/asd/tac/constellation/algorithms/clustering/hierarchical/HierarchicalControllerTopComponent.java

```diff
@@ -551,7 +551,7 @@ private void setGroups(final boolean doUpdate) {
                 dp.setState(null);
                 revalidateParents(dp);
             }
-            if (doUpdate) {
+            if ((state != null) && doUpdate) {
```

Comment:
Thanks @sol695510. Note that the additional parenthesis around the `state != null` is redundant.

Reply:
Did you want me to remove the brackets? @arcturus2

Reply:
If you could that would be great @sol695510 because I reckon Sonar will complain 😄